app/misc/git.ts

```diff
@@ -395,27 +395,29 @@ function pushToRemote() {
       displayModal("Pushing changes to remote...");
       addCommand("git push -u origin " + branch);
       repo.getRemotes()
-        .then(function (remotes) {
-          repo.getRemote(remotes[0])
-            .then(function (remote) {
-              return remote.push(
+      .then(function (remotes) {
+        repo.getRemote(remotes[0])
+        .then(function (remote) {
+          return remote.push(
                 ["refs/heads/" + branch + ":refs/heads/" + branch],
                 {
                   callbacks: {
                     credentials: function () {
-                      return cred;
+                      return getCredentials();
                     }
                   }
                 }
               );
-            })
-            .then(function () {
-              CommitButNoPush = 0;
-              window.onbeforeunload = Confirmed;
-              console.log("Push successful");
-              updateModalText("Push successful");
-              refreshAll(repo);
-            });
+        }).then(function() {
+          CommitButNoPush = 0;
+          window.onbeforeunload = Confirmed;
+          console.log("Push successful");
+          updateModalText("Push successful");
+          refreshAll(repo);
+        }).catch(function(err) {
+          console.log(err);
+          updateModalText("Push Failed : "+err.message);
```

Comment:
The error message you get if you aren't signed in is unclear "Push Failed : Method push has thrown an error.".  I see this is the error message coming from nodegit so I'm not sure if there is an easy way to fix this?

Reply:
right. but if you are not signed it, you are not supposed to push or pull right?